File: leetcode/editor/cn/1403_非递增顺序的最小子序列_test.go
package leetcode

import (
	"github.com/stretchr/testify/assert"
	"sort"
	"testing"
)

func Test_givenNormal_when1403_thenSuccess(t *testing.T) {
	assert.Equal(t, 1, 1, "shouldEqual")
}

//leetcode submit region begin(Prohibit modification and deletion)
func minSubsequence(nums []int) []int {
	sort.Sort(sort.Reverse(sort.IntSlice(nums)))
	ts, ls := 0, 0
	for _, n := range nums {
		ts += n
	}
	for idx, n := range nums {
		ls += n
		if ls > ts/2 {
			return nums[:idx+1]
		}
	}
	return []int{}
}

//leetcode submit region end(Prohibit modification and deletion)

File: leetcode/editor/cn/1403_非递增顺序的最小子序列_test.py
import unittest
from typing import List


class TestSolution(unittest.TestCase):

    def setUp(self) -> None:
        self.solution = Solution()

    def test_givenNormal_when_thenSuccess(self):
        ss = self.solution.minSubsequence([4, 3, 10, 9, 8])
        self.assertListEqual([10, 9], ss)


# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    def minSubsequence(self, nums: List[int]) -> List[int]:
        nums.sort(reverse=True)
        right, left = sum(nums) / 2, 0
        for i in range(len(nums)):
            left += nums[i]
            if left > right:
                return nums[:i + 1]


# leetcode submit region end(Prohibit modification and deletion)


if __name__ == '__main__':
    unittest.main()
